src/bootstrap/rustc.rs

```diff
@@ -83,6 +84,22 @@ fn main() {
         }
     }
 
+    // If we're compiling specifically the `panic_abort` crate then we pass the
+    // `-C panic=abort` option. Note that we do not do this for any other crate
+    // intentionally as this is the only crate for now that we ship with
+    // panic=abort.
+    //
+    // This... is a bit of a hack how we detect this. Ideally this information
+    // should be encoded in the crate I guess? Would likely require an RFC
+    // amendment to RFC 1513, however.
```

Comment:
Ah gotcha, because even when building panic_abort alone, there's the deps of libc, alloc, and core in the DAG too. Sorry for the noise.